cli/command/volume/create_test.go

```diff
@@ -192,6 +193,11 @@ func TestVolumeCreateClusterOpts(t *testing.T) {
 
 	cli := test.NewFakeCli(&fakeClient{
 		volumeCreateFunc: func(body volume.CreateOptions) (volume.Volume, error) {
+			// sort Secrets to avoid flaky test
+			secrets := body.ClusterVolumeSpec.Secrets
+			sort.SliceStable(secrets, func(i, j int) bool {
```

Comment:
Perhaps remove the intermediate variable to prevent confusing the `secrets` argument from the other `secrets` variable;

```suggestion
  			sort.SliceStable(body.ClusterVolumeSpec.Secrets, func(i, j int) bool {
```

Reply:
We should probably also update the `runCreate()` function to also sort these before making the request; https://github.com/docker/cli/blob/7f073ab823ed25ab2dfd3f800212cdab50501313/cli/command/volume/create.go#L147-L155


Wondering if we should add a `GetSlice()` (or something like that) method on `MapOpt`, which would return a sorted slice of the map; https://github.com/docker/cli/blob/c780f7c4abaf67034ecfaa0611e03695cf9e4a3e/opts/opts.go#L177-L180

Reply:
I will update the `runCreate()` function as well!

As for `getSlice()`, I think it's a good idea. It will be useful in similar testing situations and also prevent flakes in the future. Should we add it to this PR?